spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/registry/RdbmsUriRegistry.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.registry;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.sql.DataSource;
+
+import org.springframework.cloud.dataflow.server.repository.AbstractRdbmsKeyValueRepository;
+import org.springframework.cloud.dataflow.server.repository.DeploymentIdRepository;
+import org.springframework.cloud.deployer.resource.registry.UriRegistry;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.util.Assert;
+
+/**
+ * RDBMS implementation of {@link UriRegistry}.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public class RdbmsUriRegistry extends AbstractRdbmsKeyValueRepository<String> implements UriRegistry {
```

Comment:
here's the registry impl without any base class: https://github.com/markfisher/spring-cloud-dataflow/commit/47d3cd1153fc88f8eb6a8e73028e207912a264c2

tests don't pass with that as-is, presumably due to something with the test initialization script

however, it does show how the version *without* a base class actually has fewer lines of code (and does not introduce any public methods outside of the contract)